src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java

```diff
@@ -315,4 +337,30 @@ private void logStepNotInEntityScreenError(EntityScreen entityScreen, SessionDat
         neededDatum.getDataId(), step.getValue(), nodeSetString, referencesString));
 
     }
+
+    /**
+     * Execute the post request associated with the sync screen and perform a sync if necessary.
+     */
+    public void doPostAndSync(MenuSession menuSession, FormplayerSyncScreen screen) throws SyncRestoreException {
```

Comment:
nit: A Factory class doesn't seem like a great place for this public method, think it would be better to put it in it's own helper class if possible.